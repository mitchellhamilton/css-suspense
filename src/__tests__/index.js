// @flow
import * as React from "react";
import { render } from "react-dom";
import { Stylesheet } from "../";

function safeQuerySelector(selector) {
  let element = document.querySelector(selector);
  if (!element) {
    throw new Error(`Could not find element matching selector: ${selector}`);
  }
  return element;
}

function App() {
  return (
    <React.Fragment>
      {/*$FlowFixMe*/}
      <React.Placeholder delayMs={500} fallback="loading...">
        <Stylesheet href="./style.css">
          {/* this content shouldn't render
          until the stylesheet finishes loading */}
          <span className="big">what up what up</span>
        </Stylesheet>
      </React.Placeholder>
      <div>some other content</div>
    </React.Fragment>
  );
}

let sleep = time => new Promise(resolve => setTimeout(resolve, time));

test("wow", cb => {
  safeQuerySelector("body").innerHTML = `<div id="root"></div>`;
  render(<App />, safeQuerySelector("#root"), () => {
    expect(document.documentElement).toMatchSnapshot();
    safeQuerySelector("link").dispatchEvent(new Event("load"));
    setTimeout(() => {
      expect(document.documentElement).toMatchSnapshot();

      cb();
    }, 0);
  });
});
