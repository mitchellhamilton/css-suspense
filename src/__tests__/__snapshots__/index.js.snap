// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wow 1`] = `
<html>
  <head />
  <body>
    <div
      id="root"
    >
      <div>
        some other content
      </div>
    </div>
    <link
      href="./style.css"
      media="all"
      rel="stylesheet"
    />
  </body>
</html>
`;

exports[`wow 2`] = `
<html>
  <head />
  <body>
    <div
      id="root"
    >
      <link
        href="./style.css"
      />
      <span
        class="big"
      >
        what up what up
      </span>
      <div>
        some other content
      </div>
    </div>
    <link
      href="./style.css"
      media="all"
      rel="stylesheet"
    />
  </body>
</html>
`;
